cachi2/core/package_managers/yarn_classic/utils.py

```diff
@@ -0,0 +1,101 @@
+from collections import deque
+from typing import Any
+
+from pyarn.lockfile import Package as PYarnPackage
+
+from cachi2.core.package_managers.yarn_classic.project import PackageJson, YarnLock
+from cachi2.core.package_managers.yarn_classic.workspaces import Workspace
+from cachi2.core.rooted_path import RootedPath
+
+
+def find_non_dev_deps(
+    main_package_json: PackageJson,
+    yarn_lock: YarnLock,
```

Comment:
To double-check, this is a yarn.lock for the main package, right?

Reply:
yes, can you have multiple yarn.lock files in one repo ?

Reply:
Thank you, let's keep it as is then.